src/rich-text-editor.component.js

```diff
@@ -166,11 +170,23 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
   function handleSelectionChange(evt) {
     if (isFocused()) {
       const selection = window.getSelection()
-      if (selection.rangeCount > 0) {
-        selectionRangeBeforeBlurRef.current = window.getSelection().getRangeAt(0)
+      if (
+        !selection.anchorNode.isSameNode(lastSelectionAnchorNode)
+        || selection.anchorOffset !== lastSelectionAnchorOffset
+        || !selection.focusNode.isSameNode(lastSelectionFocusNode)
+        || selection.focusOffset !== lastSelectionFocusOffset
+      ) {
+        setLastSelectionAnchorNode(selection.anchorNode);
+        setLastSelectionAnchorOffset(selection.anchorOffset);
+        setLastSelectionFocusNode(selection.focusNode);
+        setLastSelectionFocusOffset(selection.focusOffset);
+        
+        if (selection.rangeCount > 0) {
+          selectionRangeBeforeBlurRef.current = window.getSelection().getRangeAt(0)
+        }
+  
+        richTextContext.fireSelectionChanged()
```

Comment:
Is there a reason this wasn't done in the first place?

Reply:
Yes -- we're trying to be a pass through of the DOM's selectionchange listener -- not adding extra logic on top of it.